### PR TITLE
libstrawberry-common: only use execinfo on glibc

### PR DESCRIPTION
**Pull request by q66**
This is a glibc-specific header, so it should only be used with glibc. This fixes compilation with musl libc and possibly other C library implementations.

### CODE REVIEW

**Comment by jonaski** [comment]
BSD's have backtrace too through the libexecinfo library. https://www.freebsd.org/cgi/man.cgi?query=backtrace&sektion=3&manpath=FreeBSD+11-current
So a better approach might be to use CMake to detect if the system has backtrace.
```
find_package(Backtrace)
set(HAVE_BACKTRACE ${Backtrace_FOUND})
```
And add `#cmakedefine HAVE_BACKTRACE` to `config.h.in`

**Comment by q66** [comment]
This also makes sure proper includes are included and proper libs are linked.

File: ext/libstrawberry-common/backtrace_inc.h.in [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Strawberry.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#cmakedefine01 Backtrace_FOUND
```
Comment (jonaski) [inline ext/libstrawberry-common/backtrace_inc.h.in]: Can this be moved to src/config.h.in ?

Comment (q66) [reply ext/libstrawberry-common/backtrace_inc.h.in]: Why? It's only ever used in one place, why pollute global includes with it?

**Pull request merged**